source/OptimaJet.DWKit.StarterApplication/Services/Workflow/WorkflowProductService.cs

```diff
@@ -184,8 +187,37 @@ public async Task ProductProcessChangedAsync(ProductProcessChangedArgs args)
             }
         }
 
+        public async Task ReassignUserTasksForProduct(Product product) {    
```

Comment:
>  In ReassignUserTasksForProduct, we would want to capture the comment from the previous UserTask(s).

I had originally done that (this commit: https://github.com/sillsdev/appbuilder-portal/pull/544/commits/59d8355cc50d22f0475aa2b7a85b96a7acb585aa#diff-d2deae5122d392cecaf0f41e802ea812R70), but ran in to a situation where the owner change may have different permissions?, so then it wouldn't make sense to just 1-1 copy over the previous user tasks. However, if that's not the case, maybe, by convention of role, we _can_ say that we can just copy over (or just change the user Id).

Reply:
The code isn't aware why the UserTasks are assigned (owner, orgadmin or others in the future).  These are determined by the available commands in the workflow.  So it isn't safe to do reassignment of tasks.  It would be good to keep the comment from the previous command execution.

Now upon reflection, the comments really should be on the completed `ProductionTransition`.  That can be fixed at a later time.  I have created https://developertown.visualstudio.com/SIL%20International%20Scriptoria/_workitems/edit/33431 to track this part.

Reply:
If it would make it cleaner to implement to save the comment to ProductionTransition, then feel free to make that change.

Reply:
It seems like there could be more shared code with just the difference in how Comments are handled.  I didn't see pulling the Comment from the previous UserTasks.

Reply:
I don't know if there is a way to pull the comment from each user task, because there is a chance for non-equal task conversion. We should start with 4 tasks and end up with 3 or 5. Is the comment the same for all tasks? (that's the only way I see carrying the comment over working :-\ )